src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1203,21 +1237,14 @@ public boolean checkOrRequestItems(final boolean useItemDamage, final boolean wa
             {
                 continue;
             }
-            final int countOfItem;
-            if (useItemDamage)
-            {
-                countOfItem = worker.getItemCountInInventory(stack.getItem(), stack.getItemDamage());
-            }
-            else
-            {
-                countOfItem = worker.getItemCountInInventory(stack.getItem(), -1);
-            }
-            if (countOfItem < 1 || (countOfItem < tempStack.stackSize && !waitForRequest))
+            final int itemDamage = useItemDamage ? stack.getItemDamage() : -1;
+            final int countOfItem = worker.getItemCountInInventory(stack.getItem(), itemDamage);
+            if (countOfItem < tempStack.stackSize)
             {
                 final int itemsLeft = stack.stackSize - countOfItem;
-                @NotNull final ItemStack requiredStack = new ItemStack(stack.getItem(), itemsLeft, -1);
+                @NotNull final ItemStack requiredStack = new ItemStack(stack.getItem(), itemsLeft, itemDamage);
 
-                if(!requiredStack.isItemEqual(getOwnBuilding().getFirstNeededItem()))
+                if(!isInNeededItems(tempStack))
```

Comment:
No, we shouldn't increase the stacksize. Since it is asynch request he only needs that amount even if he tries to get it multiple times.